### PR TITLE
feat: add vg autoindex giraffe

### PR DESCRIPTION
**Pull request by FelixMoelder**
Adding a wrapper for `vg autoindex` creating index files for the `vg giraffe` command.

### QC

* [x] I confirm that I have followed the [documentation for contributing to `snakemake-wrappers`](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).

While the contributions guidelines are more extensive, please particularly ensure that:
* [x] `test.py` was updated to call any added or updated example rules in a `Snakefile`
* [x] `input:` and `output:` file paths in the rules can be chosen arbitrarily
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`)
* [x] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to 
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command under `url:`
* [x] conda environments use a minimal amount of channels and packages, in recommended ordering

## Summary by CodeRabbit

- **New Features**
  - Introduced new configuration files for setting up a conda environment tailored for Linux 64-bit.
  - Added a new tool for creating index files for mapping reads using the `vg giraffe` command.
  - Implemented a new rule in the workflow for automated genomic data indexing.
  - Added a new genomic sequence entry in the test dataset.

- **Bug Fixes**
  - Enhanced testing capabilities for the VG toolkit's autoindexing feature.

- **Documentation**
  - Added metadata and descriptions for the new tool and its functionalities.

### CODE REVIEW

File: bio/vg/autoindex/giraffe/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+from snakemake.shell import shell
+from os import path
+
+# Extract arguments.
+extra = snakemake.params.get("extra", "")
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+prefix_path = path.splitext(snakemake.output[0])[0]
+vcf_cmd = "-v {snakemake.input['vcf']}" if snakemake.input.get("vcf", None) else ""
```
Comment (fgvieira) [inline bio/vg/autoindex/giraffe/wrapper.py:14]:
```suggestion
vcf_cmd = f"-v {snakemake.input.vcf}" if snakemake.input.get("vcf") else ""
```

Comment (FelixMoelder) [reply bio/vg/autoindex/giraffe/wrapper.py]: Thanks for adding these changes!

File: bio/vg/autoindex/giraffe/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+__license__ = "MIT"
+
+from snakemake.shell import shell
+from os import path
+
+# Extract arguments.
+extra = snakemake.params.get("extra", "")
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+prefix_path = path.splitext(snakemake.output[0])[0]
```
Comment (fgvieira) [inline bio/vg/autoindex/giraffe/wrapper.py:13]:
```suggestion
prefix_path = path.commonprefix(snakemake.output).rstrip(".")
```

File: bio/vg/autoindex/giraffe/environment.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+dependencies:
+  - vg =1.61
+  - snakemake-wrapper-utils =0.6.2
```
Comment (fgvieira) [inline bio/vg/autoindex/giraffe/environment.yaml:7]: Is this package necessary?

Comment (FelixMoelder) [reply bio/vg/autoindex/giraffe/environment.yaml]: You are right. It is not required.

**Review by fgvieira — APPROVED** [review]

**Pull request merged**